subsys/bluetooth/mesh/large_comp_data_srv.c

```diff
@@ -170,17 +170,28 @@ const struct bt_mesh_model_op _bt_mesh_large_comp_data_srv_op[] = {
 
 static int large_comp_data_srv_init(const struct bt_mesh_model *model)
 {
+	const struct bt_mesh_model *config_srv =
+		bt_mesh_model_find(bt_mesh_model_elem(model), BT_MESH_MODEL_ID_CFG_SRV);
+
 	if (!bt_mesh_model_in_primary(model)) {
 		LOG_ERR("Large Composition Data Server only allowed in primary element");
 		return -EINVAL;
 	}
 
+	if (config_srv == NULL) {
+		return -EINVAL;
+	}
+
 	/* Large Composition Data Server model shall use the device key */
 	model->keys[0] = BT_MESH_KEY_DEV;
 	model->rt->flags |= BT_MESH_MOD_DEVKEY_ONLY;
 
 	srv.model = model;
 
+	if (IS_ENABLED(CONFIG_BT_MESH_MODEL_EXTENSIONS)) {
```

Comment:
Why should we gate this with IS_ENABLED? Since extension feature is always needed, we should probably just call the API anyway. Models should have `select BT_MESH_MODEL_EXTENSIONS` in place to make this work without problems. If not, that should be fixed as a KConfig bug.